src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -754,8 +754,13 @@ private static bool Overlaps(IEnumerable<T> other, MutationInput origin)
         private static bool SetEquals(IEnumerable<T> other, MutationInput origin)
         {
             Requires.NotNull(other, nameof(other));
+            ISet<T> otherSet = other switch
+            {
+                ImmutableHashSet<T> immutableHashSet => immutableHashSet.WithComparer(origin.EqualityComparer),
```

Comment:
Why is creating a temporary `ImmutableHashSet<T>` preferable to creating a temporary `HashSet<T>`? If anything iterating over the former is going to be much slower.